clang/lib/Sema/SemaExpr.cpp

```diff
@@ -241,6 +241,19 @@ bool Sema::DiagnoseUseOfDecl(NamedDecl *D, ArrayRef<SourceLocation> Locs,
                !checkAllowedSYCLInitializer(VD, /*CheckValueDependent =*/true))
         SYCLDiagIfDeviceCode(*Locs.begin(), diag::err_sycl_restrict)
             << Sema::KernelConstStaticVariable;
+    } else if (isa<FunctionDecl>(D)) {
+      // SYCL device function cannot be called from ESIMD context. However,
+      // there are some device function declarations that are shared between
```

Comment:
Define "some"? Should we be using an allow-list of functions rather than a blanket allowance on any reserved identifier? I'm a bit concerned that this allows the user to call any builtin, intrinsic, or whatever poorly-named method they have in their own code.

Reply:
Ok, I think we can limit the set of allowed functions to those that start with `__spirv_` and `__sycl_`.

Reply:
Done.